internal/api/auth.go

```diff
@@ -129,3 +134,22 @@ func BuildJWT(user int) (string, error) {
 	// возвращаем строку токена
 	return tokenString, nil
 }
+
+func GetUserID(tokenString string) (int, error) {
+	claims := &Claims{}
+	token, err := jwt.ParseWithClaims(tokenString, claims,
+		func(t *jwt.Token) (interface{}, error) {
```

Comment:
Я бы рекомендовал использовать ключевое слово `any` вместо `interface{}` для определения любого типа, так как его как раз специально добавили для улучшения читаемости. Это алиас для  `interface{}`.

```suggestion
		func(t *jwt.Token) (any, error) {
```